pkg/filters/sets/suffixset.go

```diff
@@ -0,0 +1,63 @@
+package sets
+
+type SuffixSet struct {
+	Set    map[string]bool
+	minLen int
+	maxLen int
+}
+
+func (set *SuffixSet) Put(suffix string) {
+	if suffix == "" {
+		return
+	}
+	initLenghts := false
+	suffixLength := len(suffix)
+	if len(set.Set) == 0 {
+		set.minLen = suffixLength
+		set.maxLen = suffixLength
+		initLenghts = true
+	}
+	set.Set[suffix] = true
+
+	if !initLenghts {
+		if suffixLength < set.minLen {
+			set.minLen = suffixLength
+		}
+		if suffixLength > set.maxLen {
+			set.maxLen = suffixLength
+		}
+	}
+
+}
+
+func (set *SuffixSet) Exists(suffix string) bool {
+	return set.Set[suffix]
+}
+
+func (set *SuffixSet) Filter(val string) bool {
+	if set.minLen == 0 || set.maxLen == 0 {
+		return false
+	}
+
+	lenVal := len(val)
+
+	if lenVal < set.minLen {
+		return false
+	}
+
+	for suffixLen := set.minLen; suffixLen <= set.maxLen; suffixLen++ {
+		if lenVal < suffixLen {
+			return false
+		}
+
+		check := val[lenVal-suffixLen : lenVal]
```

Comment:
This is the only line that is different between prefixset and suffix set.
Seems to me that those two can be combined to avoid code duplication.

Reply:
True, I personally find the code simpler with separate responsibilities between the sets, but maybe there's a performance argument to be made here, so I will check that option.

Reply:
So actually, this probably wouldn't really work, because we would need to track both the prefix and suffix lengths separately.
So it would still be different loops, and the code would be less readable IMO.

Reply:
What I mean is, that you can simply pass the filter's constructor a boolean argument to describe prefix/suffix, and keep two filters for prefix or suffix. But the you can also remove one of the files and leave only one, as they are almost identical.

Reply:
I don't think it would make the code more readable, especially if internally I would need to use something like `prefixOrSuffix` everywhere, would just make it more verbose. 
I think the duplication, in this specific case, makes the codes intent more clear (this is code for suffix comparison, this is for prefix comparison).